src/modules/CRUD/components/Details/Read/AttachmentView/AttachmentView.js

```diff
@@ -0,0 +1,180 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Tab, Tabs } from '@blueprintjs/core';
+import { NavLink } from 'react-router-dom';
+import Loading from '../../../../../../components/Loading';
+import { generateURI } from '../../../../config';
+import './styles.scss';
+
+
+const buildAttachmentList = (categories, attachment) => {
+  if (!attachment || !categories) {
+    return false;
+  }
+  const { results: attachmentList } = attachment;
+  const { results: attachmentCategories } = categories;
+
+  const categoryIDsToDisplay = attachmentList.reduce((IDs, { category }) => (
+    IDs.includes(category) ? IDs : [...IDs, category]),
+  []);
+
+  return attachmentCategories
+    .filter(({ id }) => categoryIDsToDisplay.includes(id))
+    .map(category => ({
+      ...category,
+      attachments: attachmentList.filter(attach => attach.category === category.id),
+    }));
+};
+
+const AttachmentView = ({
+  attachmentCategories,
+  getAttachmentCategories,
+  attachmentCategoriesEndpoint,
+  getAttachment,
+  feature: {
+    identifier: featureId,
+    attachments: fileEndpoint,
+    pictures: pictureEndpoint,
+    attachmentFiles,
+    attachmentImages,
+  },
+  match: { params: { layer, id, action, section, category } },
+  t,
+}) => {
+  const attachmentEndpoint = section === 'attachmentFiles' ? fileEndpoint : pictureEndpoint;
+
+  const [categories, setCategories] = useState(attachmentCategories);
+  const [attachment, setAttachment] = useState({ attachmentFiles, attachmentImages });
+
+  useEffect(() => {
+    if (categories) {
+      return;
+    }
+    const getCategories = async () => {
+      const result = await getAttachmentCategories(attachmentCategoriesEndpoint);
+      setCategories(result);
+    };
+    getCategories();
+  }, [attachmentCategoriesEndpoint, categories, getAttachmentCategories]);
+
+  useEffect(() => {
+    if (attachment[section]) {
+      return;
+    }
+    const getAttachments = async () => {
+      const result = await getAttachment(attachmentEndpoint, featureId, section);
+      setAttachment({
+        ...attachment,
+        [section]: result,
+      });
+    };
+    getAttachments();
+  }, [attachment, attachmentEndpoint, featureId, getAttachment, section]);
+
+  const list = buildAttachmentList(categories, attachment[section]);
+
+  if (!list) {
+    return <Loading spinner />;
+  }
+
+  if (!list.length) {
+    return <span>t{'CRUD.details.attachment.noFiles'}.</span>;
+  }
+
+  return (
+    <Tabs
+      selectedTabId={category || list[0].name}
+      className="attachment"
+    >
+      {list.map(({ name, attachments, id: categoryId }) => (
+        <Tab
+          key={categoryId}
+          id={name}
+          title={<NavLink to={generateURI('layer', { layer, id, action, section, category: name })}>{name}</NavLink>}
+          panel={(
+            <ul className="attachment__list">
```

Comment:
This should [definitely](https://giphy.com/gifs/random-french-music-pierre-billon-11sKNq5kL1r904) be a dedicated component.

Reply:
Yep, but waiting for an UX/UI review, this component will [definitely](https://giphy.com/gifs/random-french-music-pierre-billon-11sKNq5kL1r904) be changed in the future :-)